core/src/SYCL/Kokkos_SYCL_Parallel_Team.hpp

```diff
@@ -838,26 +845,42 @@ class ParallelReduce<FunctorType, Kokkos::TeamPolicy<Properties...>,
         auto max_sg_size = kernel.get_info<
             sycl::info::kernel_device_specific::max_sub_group_size>(
             q.get_device(), sycl::range<3>(m_team_size, m_vector_size, 1));
-        if (max_sg_size % m_vector_size != 0) {
+        auto final_vector_size = std::min<int>(m_vector_size, max_sg_size);
+        if (max_sg_size % final_vector_size != 0) {
           std::stringstream out;
           out << "The maximum subgroup size (" << max_sg_size
               << ") for this kernel is not divisible by the vector_size ("
-              << m_vector_size << "). Choose a smaller vector_size!\n";
+              << final_vector_size << "). Choose a smaller vector_size!\n";
```

Comment:
same comment as above

Reply:
Yes, the used vector length is the largest power-of-two not exceeding the input vector length. The SYCL standard doesn't specify that the possible subgroup sizes are powers-of-two but that at least seems to be true for NVIDIA and Intel GPUs.
I think we can remove the check.